docs/src/main/asciidoc/mailer.adoc

```diff
@@ -36,408 +45,181 @@ The solution is located in the `mailer-quickstart` {quickstarts-tree-url}/mailer
----- 
-quarkus.mailer.auth-methods=DIGEST-MD5 CRAM-SHA256 CRAM-SHA1 CRAM-MD5 PLAIN LOGIN
-quarkus.mailer.from=YOUREMAIL@gmail.com
-quarkus.mailer.host=smtp.gmail.com
+# The SMTP host
+quarkus.mailer.host=smtp.sendgrid.net
+# The SMTP port
 quarkus.mailer.port=465
+# If the SMTP connection requires SSL/TLS
 quarkus.mailer.ssl=true
-quarkus.mailer.username=YOUREMAIL@gmail.com
-quarkus.mailer.password=YOURGENERATEDAPPLICATIONPASSWORD
-----
-
-[NOTE]
-====
-The `quarkus.mailer.auth-methods` configuration option is needed for the Quarkus mailer to support password authentication with Gmail.
-By default both the mailer and Gmail default to `XOAUTH2` which requires registering an application, getting tokens, etc.
-====
-
-== Using SSL with native executables
-
-Note that if you enable SSL for the mailer and you want to build a native executable, you will need to enable the SSL support.
-Please refer to the link:native-and-ssl[Using SSL With Native Executables] guide for more information.
-
-== Using the underlying Vert.x Mail Client
-
-The Quarkus Mailer is implemented on top of the https://vertx.io/docs/vertx-mail-client/java/[Vert.x Mail Client], providing an asynchronous and non-blocking way to send emails.
-If you need fine control on how the mail is sent, for instance if you need to retrieve the message ids, you can inject the underlying client, and use it directly:
+# Your username
+quarkus.mailer.username=....
+# Your password
+quarkus.mailer.password=....
 
-[source, java]
-----
-@Inject MailClient client;
+# By default, in dev mode, the mailer is a mock. This disables the mock and use the configured mailer.
+quarkus.mailer.mock=false
 ----
 
-Three API flavors are exposed:
-
-* the Mutiny client (`io.vertx.mutiny.ext.mail.MailClient`)
-* the bare client (`io.vertx.ext.mail.MailClient`)
-
-Check the link:vertx[Using Vert.x guide] for further details about these different APIs and how to select the most suitable for you.
-
-The retrieved `MailClient` is configured using the configuration key presented above.
-You can also create your own instance, and pass your own configuration.
+Once you have configured the mailer, if you call the HTTP endpoint as shown above, you will send emails.
 
 == Conclusion
 
-This guide has shown how you can send emails from a Quarkus application.
-The _mailer_ extension works in JVM and native mode.
+This guide has shown how to send emails from your Quarkus application.
+The xref:mailer-reference.adoc[mailer reference guide] provides more details about the mailer usage and configuration such as:
+
+* xref:mailer-reference.adoc#attachments[how to add attachments]
+* xref:mailer-reference.adoc#html[how to format the email as HTML and use inline attachments]
+* xref:mailer-reference.adoc#templates[how to use Qute templates]
+* xref:mailer-reference.adoc#testing[how to test applications sending emails]
+* xref:mailer-reference.adoc#gmail-specific-configuration[how to configure the mailer to send emails with GMAIL]
```

Comment:
I'm not sure the xrefs will be rendered properly on the website. We usually use a link (without any suffix so `link:mailer-reference#anchor`).

Reply:
It works. I've tried them before. But, do we honest, I don't know the difference. I will check.

Reply:
So, actually, we should use xref and not link. 

xref resolves correctly even when documents are included (which we want to do to build the reference guide). links are just an HTML link which instead of jumping to the right place, will target an external page.

Reply:
What I'm unsure is if they point to the correct URL on the website i.e. not having the .html suffix. I'll have a look at how the existing occurrences are rendered and let you know.

Reply:
OK, apparently what I did here https://github.com/quarkusio/quarkusio.github.io/blob/develop/_config.yml#L66 worked and we don't have the `.html` suffix when using xrefs.

Let's merge!